store/store_test.go

```diff
@@ -1707,6 +1925,52 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesAuth(c
 	c.Check(refreshDischargeEndpointHit, Equals, true)
 }
 
+func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesDeviceAuth(c *C) {
```

Comment:
it feels a bit too paranoid and annoying to maintain to have all these tests given that in the end the crucial bit is all these code paths calling the same helpers